packages/lib/src/index.ts

```diff
@@ -177,36 +228,66 @@ export default function federation(
     },
     moduleParsed(moduleInfo: ModuleInfo): void {
       for (const pluginHook of pluginList) {
-        pluginHook.moduleParsed?.call(this, moduleInfo)
+        (
+          pluginHook.moduleParsed as (
+            this: PluginContext,
+            info: ModuleInfo
+          ) => void | undefined
+        )?.call(this, moduleInfo)
       }
     },
 
     outputOptions(outputOptions) {
       for (const pluginHook of pluginList) {
-        pluginHook.outputOptions?.call(this, outputOptions)
+        (
+          pluginHook.outputOptions as
+            | ((
+                this: PluginContext,
+                options: OutputOptions
+              ) => OutputOptions | NullValue)
+            | undefined
+        )?.call(this, outputOptions)
       }
       return outputOptions
     },
 
     renderChunk(code, chunkInfo, _options) {
       for (const pluginHook of pluginList) {
-        const result = pluginHook.renderChunk?.call(
-          this,
-          code,
-          chunkInfo,
-          _options
-        )
-        if (result) {
-          return result
+        if (pluginHook.renderChunk) {
+          const result = (
+            pluginHook.renderChunk as
+              | ((
```

Comment:
Again, consider extracting the complex type to the file header to simplify this.